Modules/fcntlmodule.c

```diff
@@ -2,8 +2,8 @@
 
 // Need limited C API version 3.13 for PyLong_AsInt()
 #include "pyconfig.h"   // Py_GIL_DISABLED
-#ifndef Py_GIL_DISABLED
-#  define Py_LIMITED_API 0x030d0000
+#ifndef Py_BUILD_CORE_BUILTIN
+#  define Py_BUILD_CORE_MODULE 1
```

Comment:
You may be able to continue using the limited C API, since PyLong_AsNativeBytes() was added to the limited C API: https://github.com/python/cpython/pull/132640